### PR TITLE
Dean

### PR DESCRIPTION
**Pull request by deanmccluskey**
Added controllers, models, and routers files

### CODE REVIEW

File: controllers/categoriesController.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable prettier/prettier */
+const db = require("../models/category");
```
Comment (wparks08) [inline controllers/categoriesController.js]: Does not work as written.
```suggestion
const db = require("../models");
```
above code fixes file

File: controllers/categoriesController.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            .then(dbModel => res.json(dbModel))
+            .catch(err => res.status(422).json(err));
+    },
+    create: (req, res) => {
+        db.Category.create(req.body)
+            .then(dbModel => res.json(dbModel))
+            .catch(err => res.status(422).json(err));
+    },
+    update: (req, res) => {
+        db.Category.findOneAndUpdate({ _id: req.params.id }, req.body)
```
Comment (wparks08) [inline controllers/categoriesController.js]: Currently sends the old note (before updates occurred) back to the client. Suggest
```suggestion
        db.Category.findOneAndUpdate({ _id: req.params.id }, req.body, { new: true })
```
Adding the `new` option will send back the updated document

File: models/category.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable prettier/prettier */
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const categorySchema = new Schema({
+    date: { type: Date, default: Date.now },
+    username: { type: String, required: true },
+    category: { type: String, required: true }
+});
```
Comment (wparks08) [inline models/category.js]: Need to define a relationship between category and notes
see: [this activity from week 18]( https://ucdavis.bootcampcontent.com/ucdavis-boot-camp/UCD-SAC-FSF-PT-09-2019-U-C/blob/master/Monday_Wednesday/18-mongo-mongoose/01-Activities/18-Intro-Populate/models/Library.js)

Suggestion:
```suggestion
const categorySchema = new Schema({
    date: { type: Date, default: Date.now },
    username: { type: String, required: true },
    category: { type: String, required: true },
    notes: [
    {
      type: Schema.Types.ObjectId,
      ref: "Note"
    }
  ]

});
```

**Comment by deanmccluskey** [comment]
Thank you. Will review in AM.

**Comment by wparks08** [comment]
Looks great!

**Pull request merged**